### PR TITLE
add example of how to set up with init_app method

### PR DESCRIPTION
**Pull request by wgwz**

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -133,7 +133,21 @@ The following example initializes the extension with the standard Flask command-
     from flask_migrate import Migrate
     migrate = Migrate(app, db)
 
-The two arguments to ``Migrate`` are the application instance and the Flask-SQLAlchemy database instance. The ``Migrate`` constructor also takes additional keyword arguments, which are passed to Alembic's ``EnvironmentContext.configure()`` method. As is standard for all Flask extensions, Flask-Migrate can be initialized using the ``init_app`` method as well.
+The two arguments to ``Migrate`` are the application instance and the Flask-SQLAlchemy database instance. The ``Migrate`` constructor also takes additional keyword arguments, which are passed to Alembic's ``EnvironmentContext.configure()`` method. As is standard for all Flask extensions, Flask-Migrate can be initialized using the ``init_app`` method as well::
+
+    from flask_migrate import Migrate
+
+    migrate = Migrate()
```
Comment (miguelgrinberg) [inline docs/index.rst]: You'd want the `db` also initialized up here.

    db = SQLAlchemy()
    migrate = Migrate()

File: docs/index.rst [hunk 2]
```diff
@@ -133,7 +133,21 @@ The following example initializes the extension with the standard Flask command-
...
+
+    def create_app():
+         ...
+         # instantiate your db
```
Comment (miguelgrinberg) [inline docs/index.rst]: This should be "initialize" your db.

File: docs/index.rst [hunk 3]
```diff
@@ -133,7 +133,21 @@ The following example initializes the extension with the standard Flask command-
...
+         ...
+         migrate.init_app(app, db)
+         ...
+         # import your models
```
Comment (miguelgrinberg) [inline docs/index.rst]: The models do not necessarily need to be imported here, and a lot of people prefer to not do imports inside functions, so I would remove this comment and leave it up to the developer to decide where to do the import.

**Comment by miguelgrinberg** [comment]
Thanks for the addition, Kyle. Take a look at my comments and let me know what you think.

**Comment by wgwz** [comment]
Agreed on all points, will push up the edits later tonight.

**Comment by wgwz** [comment]
Made those edits and I added a docstring for the `create_app` function that says "Application-factory pattern". My reasoning is that this will give newcomers a term to search for if they need more info.

**Comment by miguelgrinberg** [comment]
Thanks, Kyle!

**Pull request merged**